Added support for EMR AutoScalingRole

tolis-sisiaridis opened the pull request:
AutoScalingRole association required at EMR cluster instantiation to be able to use scaling policies.

stack72 commented:
Hi @tolis-sisiaridis 

thanks for the work here - please could you enhance this PR with an acceptance test and some documentation :) Thanks!

Paul

tolis-sisiaridis commented:
Thanks Paul,

I added the option to associate an EMR cluster with a default, AWS managed IAM role ('EMR_AutoScaling_DefaultRole') as a requirement for assigning autoscaling rules.
AWS dictates for this role to be assigned at cluster creation time. I added an example to demonstrate 
how this role can be defined.

stack72 commented on builtin/providers/aws/resource_aws_emr_cluster.go:
```diff
@@ -253,12 +258,16 @@ func resourceAwsEMRClusterCreate(d *schema.ResourceData, meta interface{}) error
 
 		ReleaseLabel:      aws.String(d.Get("release_label").(string)),
 		ServiceRole:       aws.String(d.Get("service_role").(string)),
+		AutoScalingRole:   aws.String(d.Get("autoscaling_role").(string)),
```
This defined here and also, on Line 268

The declaration on line 268 is the correct one as it's optional and cannot be guaranteed to be here. We would be passing an empty string to the API with this declaration and no autoscaling role being defined

stack72 commented:
Hi @tolis-sisiaridis 

The work I asked for has been added by @jessecollier in #12823 

Thanks for the work here

Paul

tolis-sisiaridis commented:
Massive thanks to @stack72 and @jessecollier for their work!

Tolis
